src/Domain/Value/Story.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Storyblok-Api.
+ *
+ * (c) SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SensioLabs\Storyblok\Api\Domain\Value;
+
+use OskarStark\Value\TrimmedNonEmptyString;
+use Safe\DateTimeImmutable;
+use Webmozart\Assert\Assert;
+
+final readonly class Story
+{
+    public function __construct(
+        public Id $id,
+        public Uuid $uuid,
+        public ?Id $parentId,
+        public ?Uuid $groupId,
+        public string $name,
+        public string $slug,
+        public string $fullSlug,
+        public ?string $path,
+        public string $lang,
+        public int $position,
+        /**
+         * @var list<string>
+         */
+        public array $tagList,
+        public bool $isStartPage,
+        public \DateTimeImmutable $firstPublishedAt,
+        public \DateTimeImmutable $createdAt,
+        public \DateTimeImmutable $publishedAt,
+        // release_id
+        // sort_by_date
+        // meta_data
+        // alternates
+        // default_full_slug
+        // translated_slugs
+        public string $type,
+        /**
+         * @var array<string, mixed>
+         */
+        public array $content,
+    ) {
+        TrimmedNonEmptyString::fromString($this->name);
+        TrimmedNonEmptyString::fromString($this->slug);
+    }
+
+    /**
+     * @param array<string, mixed> $values
+     */
+    public static function fromArray(array $values): self
+    {
+        foreach ([
+            'id', 'uuid', 'name', 'slug', 'full_slug', 'lang', 'position', 'tag_list',
+            'is_startpage', 'first_published_at', 'created_at', 'published_at', 'content',
+        ] as $key) {
+            Assert::keyExists($values, $key);
```

Comment:
Please do not use this foreach to assert if a key exists. write this out in order of being more readable and for better and cleaner tests. and we can assert more things like not empty string etc.

Reply:
In this draft, empty string are dealt with in the constructor as it this concern more the "model check" than  the "incoming format".

Again, no problem to change this